reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/FundingSubTypeResource.java

```diff
@@ -0,0 +1,197 @@
+    this.fundingSubTypeValidator = fundingSubTypeValidator;
+  }
+
+  /**
+   * POST  /funding-sub-type : Create a new fundingType.
+   *
+   * @param fundingSubTypeDto the fundingSubTypeDto to create
+   * @return the ResponseEntity with status 201 (Created) and with body the new fundingSubTypeDTO,
+   *     or with status 400 (Bad Request) if the fundingSubType has already an ID
+   * @throws URISyntaxException if the Location URI syntax is incorrect
+   */
+  @PostMapping("/funding-sub-type")
+  @PreAuthorize("hasAuthority('reference:add:modify:entities')")
+  public ResponseEntity<FundingSubTypeDto> createFundingSubType(
+      @Valid @RequestBody FundingSubTypeDto fundingSubTypeDto) throws URISyntaxException {
+    log.debug("REST request to save FundingType : {}", fundingSubTypeDto);
+    fundingSubTypeValidator.validate(fundingSubTypeDto);
+    if (fundingSubTypeDto.getUuid() != null) {
+      return ResponseEntity.badRequest().headers(HeaderUtil
+          .createFailureAlert(ENTITY_NAME, "idexists",
+              "A new fundingSubType cannot already have an uuid")).body(null);
+    }
+    FundingSubType fundingSubType = fundingSubTypeMapper.toEntity(fundingSubTypeDto);
+    fundingSubType = fundingSubTypeService.save(fundingSubType);
+    FundingSubTypeDto result = fundingSubTypeMapper.toDto(fundingSubType);
+    return ResponseEntity.created(new URI("/api/funding-sub-type/" + result.getUuid()))
+        .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME, result.getUuid().toString()))
+        .body(result);
+  }
+
+  /**
+   * PUT  /funding-sub-type : Updates an existing fundingType.
+   *
+   * @param fundingSubTypeDto the fundingSubTypeDto to update
+   * @return the ResponseEntity with status 200 (OK) and with body the updated fundingSubTypeDto, or
+   *     with status 400 (Bad Request) if the fundingSubTypeDto is not valid, or with status 500
+   *     (Internal Server Error) if the fundingSubTypeDto couldn't be updated
+   * @throws URISyntaxException if the Location URI syntax is incorrect
+   */
+  @PutMapping("/funding-sub-type")
+  @PreAuthorize("hasAuthority('reference:add:modify:entities')")
+  public ResponseEntity<FundingSubTypeDto> updateFundingSubType(
+      @Valid @RequestBody FundingSubTypeDto fundingSubTypeDto) throws URISyntaxException {
+    log.debug("REST request to update FundingSubType : {}", fundingSubTypeDto);
+    fundingSubTypeValidator.validate(fundingSubTypeDto);
+    if (fundingSubTypeDto.getUuid() == null) {
+      return createFundingSubType(fundingSubTypeDto);
+    }
+    FundingSubType fundingSubType = fundingSubTypeMapper.toEntity(fundingSubTypeDto);
+    fundingSubType = fundingSubTypeService.save(fundingSubType);
+    FundingSubTypeDto result = fundingSubTypeMapper.toDto(fundingSubType);
+    return ResponseEntity.ok()
+        .headers(
+            HeaderUtil.createEntityUpdateAlert(ENTITY_NAME, fundingSubTypeDto.getUuid().toString()))
```

Comment:
This might happen to work, but I expect we'd would prefer to use `result.getUuid()` rahter than `fundingSubTypeDto` like in `this.createFundingSubType()`.

Reply:
Thanks. I think it was from copy and paste again. 
It's fixed and I will do more checking next time.